packages/react-ui-core/src/Menu/DropdownMenu.js

```diff
@@ -31,7 +31,11 @@ export default class DropdownMenu extends Component {
   }
 
   get selectedItem() {
-    return this.props.options[this.state.selectedIndex] || ''
+    return this.itemLabel(this.props.options[this.state.selectedIndex]) || ''
```

Comment:
Still weird to me why we need to do all this rather than just pass back the object, but :shrug:

Reply:
Perhaps the name of the function could be better. The selectedItem is only used for the dropdown anchor text.